asterius/rts/rts.mblockalloc.mjs

```diff
@@ -65,25 +85,26 @@ export class MBlockAlloc {
     return bd;
   }
 
-  freeSegment(l_end, r) {
+  freeSegment(i, l_end, r) {
     if (l_end < r) {
-      this.memory.memset(l_end, 0, r - l_end);
-      const bd = l_end + rtsConstants.offset_first_bdescr,
-            start = l_end + rtsConstants.offset_first_block,
-            blocks = (r - start) / rtsConstants.block_size;
-      this.memory.i64Store(bd + rtsConstants.offset_bdescr_start, start);
-      this.memory.i64Store(bd + rtsConstants.offset_bdescr_free, start);
-      this.memory.i64Store(bd + rtsConstants.offset_bdescr_link, 0);
-      this.memory.i32Store(bd + rtsConstants.offset_bdescr_blocks, blocks);
-      this.freeList.push(bd);
+        // memset a custom number purely for debugging help.
+        this.memory.memset(l_end, 0x42 + i, r - l_end);
```

Comment:
Better to set to a larger byte e.g. 0xff so it's more likely to get an overflowing address.

Reply:
If we memset `0xff + i`, then we can't eyeball the bit-pattern to figure out which segment it came from, unfortunately.